pkg/plugins/policies/core/rules/resourcerules.go

```diff
@@ -71,27 +71,27 @@ func WithSectionName(sectionName string) ComputeOptsFn {
 	}
 }
 
-func (rr ResourceRules) Compute(r core_model.Resource, reader ResourceReader, fn ...ComputeOptsFn) *ResourceRule {
-	opts := NewComputeOpts(fn...)
-	key := uniqueKey(r, opts.sectionName)
-	if rule, ok := rr[key]; ok {
+func (rr ResourceRules) Compute(uri UniqueResourceIdentifier, reader ResourceReader) *ResourceRule {
+	if rule, ok := rr[uri]; ok {
 		return &rule
 	}
 
-	switch r.Descriptor().Name {
+	switch uri.ResourceType {
 	case meshservice_api.MeshServiceType:
 		// find MeshService without the sectionName and compute rules for it
-		if opts.sectionName != "" {
-			return rr.Compute(r, reader)
+		if uri.SectionName != "" {
+			uriWithoutSection := copyIdentifier(uri)
```

Comment:
This is a struct with primitive field types, simple `uriWithoutSection := uri` already makes a copy

Reply:
changed